File: content/cpp/templates/try-catch.cpp
// Try-Catch Exception Handling

#include <iostream>
#include <stdexcept>
using namespace std;

double divide(double a, double b) {
    if (b == 0) {
        throw runtime_error("Division by zero!");
    }
    return a / b;
}

int main() {
    try {
        double result = divide(10, 2);
        cout << "Result: " << result << endl;
        
        result = divide(10, 0);  // This will throw
        cout << "Result: " << result << endl;
    }
    catch (const runtime_error& e) {
        cout << "Error: " << e.what() << endl;
    }
    catch (const exception& e) {
        cout << "Exception: " << e.what() << endl;
    }
    
    cout << "Program continues..." << endl;
    
    return 0;
}

File: content/java/templates/08_Files_Errors/try-catch.java
// Try-Catch Exception Handling

public class TryCatch {
    public static double divide(double a, double b) {
        if (b == 0) {
            throw new ArithmeticException("Division by zero!");
        }
        return a / b;
    }
    
    public static void main(String[] args) {
        try {
            double result = divide(10, 2);
            System.out.println("Result: " + result);
            
            result = divide(10, 0);  // This will throw
            System.out.println("Result: " + result);
        }
        catch (ArithmeticException e) {
            System.out.println("Error: " + e.getMessage());
        }
        catch (Exception e) {
            System.out.println("Exception: " + e.getMessage());
        }
        
        System.out.println("Program continues...");
    }
}
